File: Data Structures & Algorithms/implement-queue-using-stacks/submission-0.java
class MyQueue {

    private Deque<Integer> inputStack;
    private Deque<Integer> outputStack;
    private int size;

    {
        inputStack = new ArrayDeque<>();
        outputStack = new ArrayDeque<>();
    }

    public MyQueue() {
        
    }
    
    public void push(int x) {
        this.inputStack.addFirst(x);
        this.size++;
    }
    
    public int pop() {
        if (this.outputStack.isEmpty()) {
            while (!this.inputStack.isEmpty()) {
                this.outputStack.push(this.inputStack.pop());
            }
        }
        this.size--;
        return this.outputStack.pop();
    }
    
    public int peek() {
        if (this.outputStack.isEmpty()) {
            while (!this.inputStack.isEmpty()) {
                this.outputStack.push(this.inputStack.pop());
            }
        }
        return this.outputStack.peek();
    }
    
    public boolean empty() {
        return this.size == 0;    
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: Data Structures & Algorithms/implement-stack-using-queues/submission-0.py
class MyStack:

    def __init__(self):
        self.q = []
        self.top_ele = 0
        self.size = 0

    def push(self, x: int) -> None:
        self.q.append(x)
        self.top_ele = x
        self.size += 1

    def pop(self) -> int:
        for i in range(self.size - 1):
            self.top_ele = self.q[0]
            self.q.append(self.q.pop(0))
        self.size -= 1
        return self.q.pop(0)
        

    def top(self) -> int:
        return self.top_ele

    def empty(self) -> bool:
        return self.size == 0

        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()